GLTFSDK/Source/MeshPrimitiveUtils.cpp

```diff
@@ -509,6 +510,149 @@ namespace
             return rawIndices;
         }
     }
+
+    template<typename T>
+    std::vector<T> ReconstructTriangleStripIndexing(const T* indices, size_t indexCount)
+    {
+        if (indexCount % 3 != 0)
+        {
+            throw GLTFException("Input triangulated triangle strip has non-multiple-of-3 indices.");
+        }
+
+        if (indexCount < 3)
+        {
+            throw GLTFException("Input triangulated triangle strip has fewer than 3 indices.");
+        }
+
+        const size_t triangleCount = indexCount / 3;
+        const auto resultIndexCount = 2 + triangleCount;
+        std::vector<T> result(resultIndexCount);
+
+        result[0] = indices[0];
+        result[1] = indices[1];
+        result[2] = indices[2];
+
+        for (size_t i = 3; i < resultIndexCount; ++i)
+        {
+            if (i % 2 != 0)
+            {
+                result[i] = indices[3 * i - 5];
+            }
+            else
+            {
+                result[i] = indices[3 * i - 4];
+            }
+        }
+
+        return result;
+    }
+
+    template<typename T>
+    std::vector<T> ReconstructTriangleFanIndexing(const T* indices, size_t indexCount)
+    {
+        if (indexCount % 3 != 0)
+        {
+            throw GLTFException("Input triangulated triangle fan has non-multiple-of-3 indices.");
+        }
+
+        if (indexCount < 3)
+        {
+            throw GLTFException("Input triangulated triangle fan has fewer than 3 indices.");
+        }
+
+        const auto triangleCount = indexCount / 3;
+        const auto resultIndexCount = 2 + triangleCount;
+        std::vector<T> result(resultIndexCount);
+
+        result[0] = indices[0];
```

Comment:
Doing multiplication and then subtraction to get the index seems really weird to me. Why not do this, which is more like the opposite of the original transform?


```
        // indices:
        //     0,1,2
        //     0,2,3
        //     0,3,4

        result.push_back(indices[0]);
        result.push_back(indices[1]);
        for (size_t i = 2; i < triangleCount; i+=3)
        {
            result.push_back(indices[i]);
        }
```

Reply:
Fixed.